src/types.ts

```diff
@@ -48,9 +48,23 @@ export interface Component {
   type: string;
 }
 
+export const isPrefabComponent = (
+  component: PrefabReference,
+): component is PrefabComponent =>
+  typeof component.type === 'undefined' || component.type === 'COMPONENT';
+export const isPrefabPartial = (
```

Comment:
One could argue if these helpers are necessary. I think it is as clear to do these checks inline in the code, e.g.:

```typescript
if(prefabreference.type === 'PARTIAL') {
  // Code related to partial references
}
```

You don't need a user defined type guard to make sure TypeScript infers the correct type.

Reply:
I have removed the two helpers. I'm now doing a check just on the type as you have described.